### PR TITLE
Upgrade to nom v7

### PR DESCRIPTION
**Pull request by w1th0utnam3**
With respect to issue https://github.com/elrnv/vtkio/issues/26 I upgraded the legacy VTK parser to use `nom` v7 to make the crate more future proof.

For now, I mostly followed your old code directly and just "translated" it. I think for more enhancements, e.g. possibly better error handling it would be more appropriate to make new PRs.

All tests pass, however I cannot guarantee that it behaves 100% consistently compared to the old version outside of the tests (e.g. with respect to white spaces).

### CODE REVIEW

**Comment by elrnv** [comment]
Wow, this is amazing! Thank you so much @w1th0utnam3! I should have some time to review on Friday. Looking forward to it, I remember this being quite a lot of effort. If I remember correctly, I think this will also make `vtkio` quite a lot faster with legacy files, since nom has made many improvements since v3.

**Comment by w1th0utnam3** [comment]
Sounds good!

I will do some quick comparisons with larger files I have lying around.

File: src/parser.rs [hunk 1]
```diff
@@ -339,507 +322,393 @@ impl<BO: ByteOrder + 'static> VtkParser<BO> {
...
-                                    >> sz: f32_b
-                                    >> ([sx, sy, sz])
-                            )
-                        )
-                    >> data: call!(Self::attributes, ft)
-                    >> (DataSet::ImageData {
+        tagged_block(tag_no_case_line("STRUCTURED_POINTS"), |input| {
+            let (input, parms) = permutation((
+                line(array3("DIMENSIONS")),
+                line(array3("ORIGIN")),
```
Comment (elrnv) [inline src/parser.rs:498]: nice refactoring into array3!

File: src/parser.rs [hunk 2]
```diff
@@ -864,38 +733,71 @@ pub fn parse_be(input: &[u8]) -> IResult<&[u8], Vtk> {
...
+                (
+                    Version::new((2, 0)),
+                    "This is a title".to_string(),
+                    FileType::Binary
+                )
+            ))
+        );
+    }
+    #[test]
+    fn meta_test() {
```
Comment (elrnv) [inline src/parser.rs:770]: Nice addition!

File: src/parser.rs [hunk 3]
```diff
@@ -864,38 +733,71 @@ pub fn parse_be(input: &[u8]) -> IResult<&[u8], Vtk> {
...
     }
     #[test]
     fn title_test() {
         let f = title("This is a title\nBINARY".as_bytes());
-        assert_eq!(f, IResult::Done("BINARY".as_bytes(), "This is a title"));
+        assert_eq!(f, Ok(("\nBINARY".as_bytes(), "This is a title")));
+    }
+    #[test]
+    fn header_test() {
+        let f = header(" \t\n# vtk DataFile Version 2.0  \nThis is a title\nBINARY\n".as_bytes());
```
Comment (elrnv) [inline src/parser.rs:756]: Nice to see more tests with more variation!

**Review by elrnv — APPROVED** [review]
Awesome! These changes look great, more tests and good comments. Thank you once again!

**Pull request merged**